### DISCUSSION TITLE
Adding SMTP Mailgun server to Zitadel - Self Hosted

### DISCUSSION THREAD

**Discussion by filipeaaoliveira** [zitadel/zitadel#9080] [Q&A] [answered]
Hello everyone,

I'm self-hosting Zitadel and I want to connect my Mailgun to it.

I've tried it in the console, however, when testing, I get this error:

RpcError(INTERNAL): could not add smtp auth, check if both your user and password are correct, if they're correct maybe your provider requires an auth method not supported by ZITADEL (EMAIL-s9kfs)

I've reseted the password for my mailgun user, so I know that that is correct, plus, I use it somewhere else.

I've also tried to add it via the configuration, however, my zitadel is already running in docker, and I don't want to destroy the volume. I've tried to initialize it again, but without any success to add the SMTP server.

If someone can help, I would really appreciate it, as this is the last step to actually remove all our Keycloak and start using Zitadel.

Thank you.

**Comment by fforootd** [comment]
Hm what version of Zitadel do you encounter this?

I am asking because current versions of Zitadel have a template for mailgun that was working.

![image](https://github.com/user-attachments/assets/f25be12f-c26a-4200-81f6-c5827824fe06)

**Reply by filipeaaoliveira** [reply]
Hi @fforootd 

Thank you for answering this.

I'm using the latest version of Zitadel.

Yes, I always tried to integrate with the plugin, but when testing, I always get the error mentioned.

I find it strange, because I use the same configuration for my Django Project, running in the same machine and it works, however, with Zitadel, it cannot authenticate. 
I've tried to reset the password just to make sure, but I ran into the same issue.

**Reply by fforootd** [reply]
Hm what could be an issue is that Zitadel is not able to reach mailgun.

How is Zitadel deployed? Does it have an outbound connection to mailgun?

Are there any logs you can share from stdout?

**Reply by filipeaaoliveira** [reply, marked as answer]
Hello @fforootd 

Sorry to bother you, I was able to solve it.

Because our mailgun server is in EU, the default server is smtp.eu.mailgun.org, by default in Zitadel it shows smtp.mailgun.org which is correct in most cases.

Thank you for this and sorry for the time you spent on this, I really appreciate it.

I will close this discussion.

**Reply by fforootd** [reply]
Ah, thanks for sharing this. This make sense!
